lib/OpenAI/API/Request.pm

```diff
@@ -160,19 +178,13 @@ sub _request_headers {
 sub _send_request {
     my ( $self, $req ) = @_;
 
-    my $cond_var = AnyEvent->condvar;
+    my $loop = IO::Async::Loop->new();
 
-    $self->_async_http_send_request($req)->then(
-        sub {
-            $cond_var->send(@_);
-        }
-    )->catch(
-        sub {
-            $cond_var->send(@_);
-        }
-    );
+    my $future = $self->_async_http_send_request($req);
+
+    $loop->await($future);
 
-    my $res = $cond_var->recv();
+    my $res = $future->get;
```

Comment:
Also there's no need to have `await`ed above if you're just going to `->get` here. This will wait anyway.

Reply:
I tried to comment the `$loop->await($future)` line, but it caused the following error in my tests:

```
% perl -Ilib ./t/01-completions.t
Can't call method "await" on an undefined value at
/Users/nferraz/perl5/lib/perl5/IO/Async/Future.pm line 113.
```

Perhaps you meant to write it in a single line?

```perl
my $res = $loop->await($future)->get;
```